scripts/install-lsp-features/on-jupyter-server-start.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+# Install and configure assistive IDE tools *if* running high enough JupyterLab version (v3)
+#
+# This script sets up code completion, continuous hinting, hover tips, code formatting, markdown
+# spell-checking, and an S3 bucket explorer UI. See below comments for details.
+
+set -eu
+
+# Minimum required JupyterLab version
+# Per https://jupyterlab-lsp.readthedocs.io/en/latest/Installation.html
+MIN_JLAB_VERSION='3.1.0'
+
+# Function to compare two X.Y.Z semantic version strings
+# Returns 0 if equal, 1 if A > B, 2 if A < B.
+# From https://stackoverflow.com/a/4025065
+compare_semvers () {
+    if [[ $1 == $2 ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
+        then
+            return 1
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            return 2
+        fi
+    done
+    return 0
+}
+
+echo "Checking conda environments"
+if conda info --envs | grep ^studio; then
+    # Standard on JLv3 image at time of writing
+    CONDA_ENV=studio
+else
+    # Standard on JLv1 image at time of writing
+    CONDA_ENV=base
+fi
+echo "Activating conda env $CONDA_ENV"
+source activate $CONDA_ENV
+
+JLAB_VERSION=`jupyter lab --version`
+echo "Found JupyterLab version $JLAB_VERSION"
+set +e
+compare_semvers $MIN_JLAB_VERSION $JLAB_VERSION
+VERCOMP_RESULT=$?
+set -e
```

Comment:
So our official documentation recommends an env variable check to do this:
https://docs.aws.amazon.com/sagemaker/latest/dg/studio-jl.html#studio-jl-install

I'm torn because SageMaker JL3 will always be > 3.1; but there is a future where JL4 would require an update to these scripts if we use the env check.

Other than matching official documentation, my issue with this code block is that its 50 lines of code and there is a limit for number of characters that can be serialized in a LCC. Combining his script with others runs the risk of hitting that limit:
https://docs.aws.amazon.com/sagemaker/latest/dg/notebook-lifecycle-config.html

I think I'd prefer to match our official docs, but if you feel strongly about a true semver check then I'm alright with it.

Reply:
Raised doc feedback to tackle the broken snippet on that page (`conda deactivate studio` is not a command) and aligned sample to match the doc.

It seems LCC limit is 16384 characters for both [notebook instances](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateNotebookInstanceLifecycleConfig.html) and [SageMaker Studio](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateStudioLifecycleConfig.html#API_CreateStudioLifecycleConfig_RequestSyntax) per the docs.

It seems a bit opaque to rely on the magic env var, but agree script length & simplicity is important.